src/secrets/components/SecretsTab.tsx

```diff
@@ -0,0 +1,178 @@
+      icon={IconFont.Plus}
+      onClick={handleOpenCreateOverlay}
+      testID="button-create"
+    />
+  )
+
+  const SecretsEmptyState = (): JSX.Element => {
+    if (!searchTerm) {
+      return (
+        <>
+          <EmptyState size={ComponentSize.Medium}>
+            <EmptyState.Text>
+              Looks like there aren't any <b>Secrets</b>, why not create one?
+            </EmptyState.Text>
+            {createSecretButton}
+          </EmptyState>
+        </>
+      )
+    }
+
+    return (
+      <>
+        <EmptyState size={ComponentSize.Medium}>
+          <EmptyState.Text>No Secrets match your query</EmptyState.Text>
+        </EmptyState>
+      </>
+    )
+  }
+
+  const handleFilterChange = (searchTerm: string) => {
+    handleFilterUpdate(searchTerm)
+  }
+
+  const handleFilterUpdate = (searchTerm: string) => {
+    setSearchTerm(searchTerm)
+  }
+
+  const handleSort = (
+    sortKey: SecretSortKey,
+    sortDirection: Sort,
+    sortType: SortTypes
+  ): void => {
+    setSortDirection(sortDirection)
+    setSortKey(sortKey)
+    setSortType(sortType)
+  }
+
+  // const handleOpenImportOverlay = (): void => {
+  // const {history, match} = props
+  //
+  // history.push(`/orgs/${match.params.orgID}/settings/variables/import`)
+  // }
+
+  // const handleOpenCreateOverlay = (): void => {
+  // const {history, match} = props
+  //
+  // history.push(`/orgs/${match.params.orgID}/settings/variables/new`)
+  // }
+
+  // const handleDeleteSecret = (secret: Secret): void => {
+  // const {onDeleteSecret} = props
+  // onDeleteSecret(secret.id)
+  // }
+
+  const leftHeaderItems = (
+    <>
+      <SearchWidget
+        placeholderText="Filter secrets..."
+        searchTerm={searchTerm}
+        onSearch={handleFilterChange}
+      />
+      <ResourceSortDropdown
+        onSelect={handleSort}
+        resourceType={ResourceType.Secrets}
+        sortDirection={sortDirection}
+        sortKey={sortKey}
+        sortType={sortType}
+      />
+    </>
+  )
+
+  return (
+    <>
+      <TabbedPageHeader
+        childrenLeft={leftHeaderItems}
+        childrenRight={createSecretButton}
+      />
+      <GetResources resources={[ResourceType.Secrets]}>
+        <FilterSecrets
+          list={secrets}
+          searchTerm={searchTerm}
+          searchKeys={['key', 'secret[].key']}
+        >
+          {sc => (
+            <SecretsList
+              secrets={sc}
+              emptyState={SecretsEmptyState()}
+              onDeleteSecret={() => {}}
+              sortKey={'key'}
+              sortDirection={sortDirection}
+              sortType={sortType}
+            />
+          )}
+        </FilterSecrets>
+      </GetResources>
+    </>
+  )
+}
+
+const mstp = (state: AppState) => {
+  const secrets = getAll<Secret>(state, ResourceType.Secrets)
+
+  return {secrets}
+}
+
+// const mdtp = {
+//     onDeleteSecret: deleteSecret,
+// }
+
+const connector = connect(mstp, null)
+
+export default connector(SecretsTab)
```

Comment:
Let's try and stick the hook pattern here since that's got way less baggage

Reply:
Yeah after refactoring this out a few times in this PR now I completely agree with that assessment.